src/Raven.Client/Documents/Session/Operations/BatchOperation.cs

```diff
@@ -365,6 +369,44 @@ private void HandleDeleteInternal(BlittableJsonReaderObject batchResult, Command
             }
         }
 
+        private void HandleForceRevisionCreation(BlittableJsonReaderObject batchResult)
+        {
+            // Need to update the change-vector of tracked entities in the session
+            // because ForceRevisionCreation can also create (put) new documents in the server:
+            // when forcing a revision for a document that did not have any revisions yet (original document is updated/saved with has-revisions flag)
+            
+            if (GetBooleanField(batchResult, CommandType.ForceRevisionCreation, "RevisionCreated") == false)
+            { 
+                // no forced revision was created...nothing to update.
+                return;
+            }
+            
+            var id = GetLazyStringField(batchResult, CommandType.ForceRevisionCreation, Constants.Documents.Metadata.Id);
```

Comment:
Isn't this code duplicate from `HandleEntityPut`, can we avoid this duplication?